lib/entities/SerializableEntity.js

```diff
@@ -0,0 +1,14 @@
+class SerializableEntity {
+    /**
+     *
+     * @param options {object}
+     * @param [options.metaData] {object}
```

Comment:
I would prefer fixed types instead of that loose metadata objects to support proper type checking.
This also applies to the other classes.

Reply:
While agree that being statically typed does provide a lot of benefits, I think having static types with dynamic additional properties gives us the best of both worlds.

Nearly everything is still getting type-checked but we also have a lot of flexibility there.

For example this roborock specific quirk:
https://github.com/Hypfer/Valetudo/blob/bc29f4eed26516682bb91e3771f3346902680e5b/lib/devices/Roborock.js#L144-L147
Without dynamic addional properties, this would've required either a new Class which inherits from StatusStateAttribute or some other hack. Neither of those being desirable since it's just for this one vendor-specific quirk.

I also like to think about the metaData property as some kind of staging area which's contents may be refactored into a new sub-class when there is enough demand and/or knowledge about them gathered.

Looking at the PointMapEntity,
https://github.com/Hypfer/Valetudo/blob/bc29f4eed26516682bb91e3771f3346902680e5b/lib/RRMapParser.js#L418-L428

i really don't want to have a `RobotPositionPointMapEntity` which is just the `PointMapEntity` with an `angle` property. However I also don't want to bloat the `PointMapEntity` with an `angle` property since that is specific to this one use-case:
Maybe at some point there are multiple PointMapEntities with an angle. Then, this would get refactored

Interestingly, Shopify does have a similar concept to make it easier for different third-party shopify integrations to interact with each other or add additional information without needing to change a schema:
https://shopify.dev/docs/admin-api/rest/reference/metafield?api[version]=2020-04